src/esm_runscripts/workflow.py

```diff
@@ -1,49 +1,189 @@
 import sys, copy, os
 import esm_parser
 
+#import pdb
+
+class Workflow:
+    """A workflow class."""
+    default_phases = []
+    user_phases = []
+    always_run_with = []
+    first_task_in_queue = ""
+    last_task_in_queue = ""
+    next_run_triggered_by = ""
+
+    def __init__(self, phases, always_run_with=[]):
+        """
+        Create a new workflow.
+
+        Arguments:
+            phases -- List of workflow phases
+            always_run_with -- List of phases that precedes each phase
+        """
+        # TODO: NW call here the phase object ???
+        self.phases = phases
+        self.always_run_with = always_run_with
+
+    def num_phases_in_workflow(self):
+        """
+        Return the number of phases in workflow.
+        """
+        return len(self.phases)
+
+    def write_to_config(self, config):
+        """
+        Write to config.
+        """
+        # NW: It is assumed here, that there are no workflows in config["general"]
```

Comment:
```suggestion
        # It is assumed here, that there are no workflows in config["general"]
```
We can see how wrote the comment with git blame, so at least Paul and myself had until now the agreement that no person is to be named in the comments unless we are doing something risky. Totally up to you whether you decide to remove it or not, just so you are aware of what our practices are.